core/src/main/scala/kafka/network/RequestChannel.scala

```diff
@@ -181,11 +183,12 @@ object RequestChannel extends Logging {
 
       if (isRequestLoggingEnabled) {
         val detailsEnabled = requestLogger.underlying.isTraceEnabled
-        val responseString =
-          if (response.responseSend.isDefined)
-            response.responseAsString.getOrElse(
+        val responseString = response match {
```

Comment:
Can we give `Response` a `responseString` method and skip this matching? The default implementation can return an empty string.